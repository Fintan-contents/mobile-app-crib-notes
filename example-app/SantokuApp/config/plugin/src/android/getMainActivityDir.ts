import {AndroidConfig} from '@expo/config-plugins';
import path from 'path';

export async function getMainActivityDir(projectRoot: string) {
  const mainActivity = await AndroidConfig.Paths.getMainActivityAsync(projectRoot);
  return path.dirname(mainActivity.path);
}
