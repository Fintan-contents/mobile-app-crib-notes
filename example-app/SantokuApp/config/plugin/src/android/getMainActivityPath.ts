import {AndroidConfig} from '@expo/config-plugins';

export async function getMainActivityPath(projectRoot: string) {
  const mainActivity = await AndroidConfig.Paths.getMainActivityAsync(projectRoot);
  return mainActivity.path;
}
