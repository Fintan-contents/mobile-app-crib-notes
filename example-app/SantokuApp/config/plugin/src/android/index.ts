export * from './withAndroidAddAppActivity';
export * from './withAndroidAddAppActivityAndroidManifest';
export * from './withAndroidAddLicensePlugin';
export * from './withAndroidAppBuildGradleForRelease';
export * from './withAndroidCopyTemplateJavaFile';
export * from './withAndroidRemoveUsesClearTextTrafficForRelease';
export * from './withAndroidDisabledWindowDrawsSystemBarBackgrounds';
export * from './withAndroidEnabledStatusBarTranslucent';
export * from './withAndroidCopyMainActivity';
export * from './withAndroidMoveDevSettingsActivityToDebugAndroidManifest';
export * from './withAndroidCopySplashScreenImages';
export * from './withAndroidCopyTemplateSplashScreenXml';
export * from './withAndroidFlexibleSplashScreen';
