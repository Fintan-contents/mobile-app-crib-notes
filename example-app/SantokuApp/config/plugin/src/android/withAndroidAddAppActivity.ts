import {ConfigPlugin, withPlugins} from '@expo/config-plugins';

import {withAndroidAddAppActivityAndroidManifest} from './withAndroidAddAppActivityAndroidManifest';
import {withAndroidCopyMainActivity} from './withAndroidCopyMainActivity';
import {withAndroidCopyTemplateJavaFile} from './withAndroidCopyTemplateJavaFile';
import {ANDROID} from '../constants';

const MAIN_ACTIVITY_FILE_NAME = 'MainActivity.java';
const APP_ACTIVITY_CLASS_NAME = 'AppActivity';

/**
 * Transition中にActivityのbackgroundが表示される問題があるため、Splash Screen用とReact Native用にActivityを分けます。
 * https://github.com/software-mansion/react-native-screens/issues/380
 *
 * 1. 「MainActivity.java」を「AppActivity.java」というファイル名でコピー（併せて、「AppActivity.java」のクラス名を「AppActivity」に変更）
 * 2. テンプレートのMainActivity.javaを「android/app/src/main/[パッケージ名]」にコピー（既存のMainActivity.javaを上書き）
 * 3. AndroidManifest.xmlにAppActivityを追加
 *
 * 1でMainActivity.javaを単純にリネームしないでコピーしているのは、MainActivity.javaが存在しないとExpo prebuild中にエラーが発生するためです。
 * @param config ExpoConfig
 */
export const withAndroidAddAppActivity: ConfigPlugin = config => {
  return withPlugins(config, [
    [
      withAndroidCopyTemplateJavaFile,
      {srcDir: ANDROID.PLUGIN_TEMPLATE_MAIN_ACTIVITY_DIR, fileName: MAIN_ACTIVITY_FILE_NAME},
    ],
    [withAndroidCopyMainActivity, {toClassName: APP_ACTIVITY_CLASS_NAME}],
    withAndroidAddAppActivityAndroidManifest,
  ]);
};
