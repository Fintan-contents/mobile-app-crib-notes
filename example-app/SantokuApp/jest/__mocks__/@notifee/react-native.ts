import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
