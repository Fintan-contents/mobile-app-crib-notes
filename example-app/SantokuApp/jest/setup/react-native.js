// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
