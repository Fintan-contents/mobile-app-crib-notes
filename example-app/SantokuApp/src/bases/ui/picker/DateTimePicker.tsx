import {log} from 'bases/logging';
import React from 'react';
import {Platform, TextInputProps as RNETextInputProps, ViewProps} from 'react-native';

import {
  DateTimePickerItemsAndroidProps,
  DateTimePickerItemsIOSProps,
  DateTimePickerItemsProps,
} from './DateTimePickerItems';
import {DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdropProps} from './PickerBackdrop';
import {PickerContainerProps} from './PickerContainer';

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;

export type PickerItemsProps =
  | Omit<DateTimePickerItemsIOSProps, 'value' | 'onChange' | 'mode' | 'display' | 'maximumDate' | 'minimumDate'>
  | Omit<DateTimePickerItemsAndroidProps, 'value' | 'onChange' | 'mode' | 'display' | 'maximumDate' | 'minimumDate'>;

export type DateTimePickerProps = {
  /**
   * 選択されたアイテムのKey
   */
  selectedValue?: Date;
  /**
   * 選択されたアイテムが存在しない場合のデフォルト値
   */
  defaultValue?: Date;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (date?: Date) => void;
  /**
   * [iOS]
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * [Android]
   * PickerのBackdrop領域をタップして閉じた場合、または「キャンセル」ボタンをタップして閉じた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   */
  onDismiss?: (date?: Date) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (date?: Date) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   * @platform ios
   */
  onCancel?: (date?: Date) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (date?: Date) => void;
  /**
   * NeutralButtonがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * @platform android
   */
  onNeutralButtonPressed?: (date?: Date) => void;
  /**
   * 選択した日時のフォーマッタ
   */
  formatText?: (date?: Date) => string | undefined;
  /**
   * 選択可能な最大日時
   * @see https://github.com/react-native-datetimepicker/datetimepicker#maximumdate-optional
   */
  maximumDate?: Date;
  /**
   * 選択可能な最小日時
   * @see https://github.com/react-native-datetimepicker/datetimepicker#minimumdate-optional
   */
  minimumDate?: Date;
  /**
   * 時刻の表示を24時間表記にするかどうか
   * @see https://github.com/react-native-datetimepicker/datetimepicker#is24hour-optional-windows-and-android-only
   * @platform android
   */
  is24Hour?: boolean;
  /**
   * Pickerのモード
   * @see https://github.com/react-native-datetimepicker/datetimepicker#mode-optional
   */
  mode?: DateTimePickerItemsProps['mode'];
  /**
   * Pickerの表示方法
   * このコンポーネントでは、spinnerとinlineのみ使用できます。
   * @see https://github.com/react-native-datetimepicker/datetimepicker#display-optional
   * @platform ios
   */
  displayIOS?: Exclude<DateTimePickerItemsProps['display'], 'clock' | 'calendar' | 'compact' | 'default'>;
  /**
   * Pickerの表示方法
   * @see https://github.com/react-native-datetimepicker/datetimepicker#display-optional
   * @platform android
   */
  displayAndroid?: Exclude<DateTimePickerItemsProps['display'], 'compact' | 'inline'>;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   * @platform ios
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   * @platform ios
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   * @platform ios
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   * @see https://github.com/react-native-datetimepicker/datetimepicker#component-props--params-of-the-android-imperative-api
   * @platform ios
   */
  pickerItemsProps?: PickerItemsProps;
  /**
   * PickerBackdropのProps
   * @platform ios
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   * @platform ios
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   * @platform ios
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

export const DateTimePicker = (props: DateTimePickerProps) => {
  React.useEffect(() => {
    log.warn(`DateTimePicker is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
