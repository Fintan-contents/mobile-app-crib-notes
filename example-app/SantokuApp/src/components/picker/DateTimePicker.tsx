import {log} from 'framework/logging';
import React from 'react';
import {Platform, TextInputProps as RNETextInputProps, ViewProps} from 'react-native';

import {DateTimePickerItemsProps} from './DateTimePickerItems';
import {DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdropProps} from './PickerBackdrop';
import {PickerContainerProps} from './PickerContainer';

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;

export type DateTimePickerProps = {
  /**
   * 選択されたアイテムのKey
   */
  selectedValue?: Date;
  /**
   * 選択されたアイテムが存在しない場合のデフォルト値
   */
  defaultValue?: Date;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (date?: Date) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   */
  onDismiss?: (date?: Date) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (date?: Date) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onCancel?: (date?: Date) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (date?: Date) => void;
  /**
   * 選択した日時のフォーマッタ
   */
  formatText?: (date?: Date) => string | undefined;
  /**
   * 選択可能な最大日時
   * @see https://github.com/react-native-datetimepicker/datetimepicker#maximumdate-optional
   */
  maximumDate?: Date;
  /**
   * 選択可能な最小日時
   * @see https://github.com/react-native-datetimepicker/datetimepicker#minimumdate-optional
   */
  minimumDate?: Date;
  /**
   * 時刻の表示を24時間表記にするかどうか
   * @see https://github.com/react-native-datetimepicker/datetimepicker#is24hour-optional-windows-and-android-only
   */
  is24Hour?: boolean;
  /**
   * Pickerのモード
   * @see https://github.com/react-native-datetimepicker/datetimepicker#mode-optional
   */
  mode?: DateTimePickerItemsProps['mode'];
  /**
   * Pickerの表示方法
   * このコンポーネントでは、spinnerとinlineのみ使用できます。
   * @see https://github.com/react-native-datetimepicker/datetimepicker#display-optional
   * @platform ios
   */
  displayIOS?: Exclude<DateTimePickerItemsProps['display'], 'clock' | 'calendar' | 'compact' | 'default'>;
  /**
   * Pickerの表示方法
   * @see https://github.com/react-native-datetimepicker/datetimepicker#display-optional
   * @platform android
   */
  displayAndroid?: Exclude<DateTimePickerItemsProps['display'], 'compact' | 'inline'>;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   * @see https://github.com/react-native-datetimepicker/datetimepicker#component-props--params-of-the-android-imperative-api
   */
  pickerItemsProps?: Omit<
    DateTimePickerItemsProps,
    'value' | 'onChange' | 'mode' | 'display' | 'maximumDate' | 'minimumDate'
  >;
  /**
   * PickerBackdropのProps
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

export const DateTimePicker = (props: DateTimePickerProps) => {
  React.useEffect(() => {
    log.warn(`DateTimePicker is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
