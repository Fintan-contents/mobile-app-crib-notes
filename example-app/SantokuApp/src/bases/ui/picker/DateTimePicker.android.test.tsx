import {render, screen} from '@testing-library/react-native';
import React from 'react';

import {DateTimePicker} from './DateTimePicker.android';

// DateTimePickerAndroidを使用した場合、自動テストではDateTimePickerAndroid.openを実行しても何もレンダリングされないため、
// 自動テストではopenする前の初期表示のみテスト対象とする
describe('DateTimePicker only with required props', () => {
  it('renders successfully if invisible', () => {
    render(<DateTimePicker textInputProps={{testID: 'textInput'}} pickerItemsProps={{testID: 'pickerItems'}} />);

    expect(screen).toMatchSnapshot('DateTimePicker if invisible.');

    const textInput = screen.queryByTestId('textInput');
    const pickerItems = screen.queryByTestId('pickerItems');
    expect(textInput).not.toBeNull();
    expect(pickerItems).toBeNull();
  });
});
