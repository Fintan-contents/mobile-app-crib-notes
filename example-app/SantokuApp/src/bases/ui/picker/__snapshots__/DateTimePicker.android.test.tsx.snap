// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker only with required props renders successfully if invisible: DateTimePicker if invisible. 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="pressableContainer"
>
  <View
    pointerEvents="box-only"
  >
    <TextInput
      editable={false}
      testID="textInput"
    />
  </View>
</View>
`;
